### ISSUE TITLE
[CP][stable channel] Use uris for paths in git source descriptions

### ISSUE THREAD

**Issue by Sese-Schneider** [dart-lang/sdk#47186]
**Commit to merge:** https://github.com/dart-lang/pub/commit/9f2ddc02f066c68fb638ff65fbb73761ece23267 - I don't know the exact process of how you bundle the [pub](https://github.com/dart-lang/pub) tool into the SDK. If this is the wrong place to file an issue, please let me know 🙂 

**Merge instructions:** Bundle pub tool into dart SDK.

**What is the issue:** When using `dart pub run build_runner build` (using the [build_runner package](https://pub.dev/packages/build_runner)), the command immediately crashed because the lock files paths are generated incorrectly. See https://github.com/dart-lang/pub/issues/3012 for more details.

**Why cherrypick:** This issue does affect the stable versions of Dart and Flutter. Using `build_runner` with *git* dependencies is not possible anymore. This should be hotfixed, or at least in the next version of the SDKs.

**Risk:** Low?. Uses uris instead of strings for the paths. Changes were covered by tests.

**Links to original issues:** 
- https://github.com/dart-lang/pub/pull/3063
- https://github.com/dart-lang/pub/issues/3012
- https://github.com/flutter/flutter/issues/89752

**Comment by athomas** [comment]
@jonasfj @devoncarew

**Comment by devoncarew** [comment]
Thanks for the cherry pick request!

@jonasfj - can you confirm this was an issue and is in the current stable? It would also be good to know things like the risk to cherry pick (can it be cherry picked in cleanly, ...) and the relative impact of the issue - how many users would be affected, the platforms, ... Thanks!

**Comment by jonasfj** [comment]
The risk from cherrypick is small, it would drag along [0f3055db](https://github.com/dart-lang/pub/commit/0f3055db707a338decdd963a29edcbfd9fbb8b39) which:
 * only affects publishing, and,
 * only for packages that have an `executables` section, which would now be validated.

------------

The changes from 70b1a4 to 9f2ddc are:

```
git log --format="%C(auto) %h %s" 70b1a4f9229a36bac6340ec7eae2b2068baac96c..9f2ddc02f066c68fb638ff65fbb73761ece23267
 9f2ddc02 Use uris for paths in git source descriptions (#3063)
 97a0033e Update README.md (#3069)
 2ef10adf Roll setup-dart (#3054)
 4b5450d3 Updated repository specification. (#3050)
 0f3055db Add executables to allErrors (#3048)
 ```

--------------------------------------

This affects users who are:
 * Running Windows,
 * Have a git-dependency where `pubspec.yaml` is in a subfolder of the git-repository, and,
 * Run `dart run <package>` (I'm guessing, not sure why it's only reported for `dart run build_runner`).

**Comment by sigurdm** [comment]
Cherry pick is here: https://dart-review.googlesource.com/c/sdk/+/213265

**Comment by devoncarew** [comment]
Thanks for the info! We'll review Tues.

**Comment by devoncarew** [comment]
Approved; cc @athomas.

**Comment by athomas** [comment]
Merged to stable in 3300f32fdc5432f40bc00f4179529cbd7449c93a (`2.14.2`).

**Comment by Sese-Schneider** [comment]
Thank you very much for the fast progress here 😍🚀 
@athomas @devoncarew @sigurdm @jonasfj ❤️‍🔥
